lib/models/recs.js

```diff
@@ -0,0 +1,154 @@
+    return array.concat(element);
+  });
+  const dataArray2Reduced = dataArray2.reduce(function (array, element) {
+    return array.concat(element);
+  });
+
+  const joinedArrays = dataArray1Reduced.concat(dataArray2Reduced);
+
+  const counts = {};
+  joinedArrays.forEach(function (x) {
+    counts[x] = (counts[x] || 0) + 1;
+  });
+
+  const keysSortedDesc = Object.keys(counts).sort(function (a, b) {
+    return counts[b] - counts[a];
+  });
+
+  const topThreeKeys = [
+    keysSortedDesc[0],
+    keysSortedDesc[1],
+    keysSortedDesc[2],
+  ];
+  return topThreeKeys;
+}
+
+async function getMostWatchedGenres(userId) {
+  const user = await User.findById(userId);
+  if (!user) {
+    return 'User not found';
+  } else if (!user.towatch || !user.watched) {
+    return null;
+  } else {
+    const toWatchListAllGenres = user.towatch.map((show) => {
+      const genres = show.genres_id.map((genre) => {
+        return genre;
+      });
+      return genres;
+    });
+
+    const watchedListAllGenres = user.watched.map((show) => {
+      const genres = show.genres_id.map((genre) => {
+        return genre;
+      });
+      return genres;
+    });
+
+    const mostWatchedGenres = calculateTopThree(
+      toWatchListAllGenres,
+      watchedListAllGenres
+    );
+    return mostWatchedGenres;
+  }
+}
+
+async function getMostWatchedNetworks(userId) {
+  const user = await User.findById(userId);
+  if (!user) {
+    return 'User not found';
+  } else if (!user.towatch || !user.watched) {
+    return null;
+  } else {
+    const toWatchListAllNetworks = user.towatch.map((show) => {
+      const networks = show.networks.map((network) => {
+        return network.id;
+      });
+      return networks;
+    });
+
+    const watchedListAllNetworks = user.watched.map((show) => {
+      const networks = show.networks.map((network) => {
+        return network.id;
+      });
+      return networks;
+    });
+
+    const mostWatchedNetworks = calculateTopThree(
+      toWatchListAllNetworks,
+      watchedListAllNetworks
+    );
+
+    return mostWatchedNetworks;
+  }
+}
+
+async function getAllWatchlistShows(userId) {
+  const user = await User.findById(userId);
+  const toWatchListAllShowsById = user.towatch.map((show) => {
+    return show.id;
+  });
+
+  const watchedListAllShowsById = user.watched.map((show) => {
+    return show.id;
+  });
+
+  const allWatchListShowsById = toWatchListAllShowsById.concat(
+    watchedListAllShowsById
+  );
+  return allWatchListShowsById;
+}
+
+async function getRecsFromAPI(data, userId, filter) {
+  const response = await fetch(
+    ` https://api.themoviedb.org/3/discover/tv?api_key=${
+      process.env.TMDB_API_KEY
+    }&language=en-US&sort_by=popularity.desc&page=1&timezone=America%2FNew_York&with_${filter}=${
+      data[0]
+    }${filter === 'genres' ? '%2C' : '%7C'}${data[1]}${
+      filter === 'genres' ? '%2C' : '%7C'
+    }${data[2]}&include_null_first_air_dates=false`
+  );
```

Comment:
What about this part? `filter === 'genres' ? '%2C' : '%7C'`
You can create a new variable for this to make it easier to understand.